app/models/reference_payment_settings.rb

```diff
@@ -31,4 +33,16 @@ def payment_link_url_present?
   def payment_link_checked?
     payment_link == '1'
   end
+
+  def saved_payment_link_url
+    return ServiceConfiguration.find_by(service_id: service_id, name: 'PAYMENT_LINK')&.decrypt_value if payment_link_url.nil?
```

Comment:
```suggestion
    return ServiceConfiguration.find_by(service_id: service_id, name: 'PAYMENT_LINK')&.decrypt_value if payment_link_url.blank?
```

`blank?` will also cover an empty string as well as nil

Reply:
It is nil in case the user input en empty string '' in the payment link url field